### PR TITLE
Add `write-file-atomically` types

### PR DESCRIPTION
**Pull request by Aankhen**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/write-file-atomically/write-file-atomically-tests.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import * as writeFileAtomically from 'write-file-atomically';
```
Comment (RyanCavanaugh) [inline types/write-file-atomically/write-file-atomically-tests.ts]: Classes and functions cannot be imported using `import * as` syntax. Please fix either the test or the definition file to use a correct import form.

See also https://stackoverflow.com/questions/39415661/what-does-resolves-to-a-non-module-entity-and-cannot-be-imported-using-this

**Review by RyanCavanaugh — CHANGES_REQUESTED** [review]

File: types/write-file-atomically/write-file-atomically-tests.ts (hunk 1 above)
Comment (Aankhen) [reply types/write-file-atomically/write-file-atomically-tests.ts]: I see. Thanks for the link. Strange that neither the lint nor the testing caught it.

Should I be using `export default` in this case? Or is it reasonable to expect users to say `import writeFileAtomically = require("write-file-atomically");`?

Comment (RyanCavanaugh) [reply types/write-file-atomically/write-file-atomically-tests.ts]: Does the module actually provide a `default` export, or is it a CommonJS module that you could only import through `require` ?

**Comment by RyanCavanaugh** [comment]
Thanks!

**Pull request merged**